[docs] CToSwift: Add section on methods

jrose-apple opened the pull request:
There's a lot here, so I might have missed something, but I think this document is complete!

jrose-apple committed:
[docs] CToSwift: Add section on methods

There's a lot here, so I might have missed something, but I think this
document is complete!

jrose-apple commented:
@swift-ci Please smoke test

varungandhi-apple commented on docs/CToSwiftNameTranslation.md:
```diff
@@ -212,6 +212,244 @@ Additionally, typedefs for `void *` or `const void *` that are themselves annota
+
+5. Other methods are imported as plain methods.
+
+If a method overrides a superclass or adopted protocol method, the Swift name of the overridden method will be used for consistency. If there's more than one such name, one is chosen arbitrarily.
```
Could you give a small example of how the case of "more than one such name" would arise? Can a single Obj-C method override multiple methods at once?

jrose-apple replied:
I guess I can put an example in. It's mostly just `NS_SWIFT_NAME` on one protocol requirement and not on another requirement in a different protocol with the same selector. It's such a terrible situation that I don't want to spend much time on it. :-(

varungandhi-apple replied:
I just asked for my own curiosity. I didn't realize you could override multiple methods at once. You don't need to put it in if you think it doesn't warrant discussion.

jrose-apple replied:
"override" is a fuzzy term here; it's being used for satisfying protocol requirements too. Maybe that's what should be clarified.

jrose-apple replied:
(Clang uses this terminology, but normal Objective-C programmers don't.)

jrose-apple committed:
[docs] CToSwift: Tweak wording around "overrides"

The previous wording referred to a method or property satisfying a
protocol requirement as an "override", which both Clang and the Swift
Clang Importer use internally sometimes. But most people who program
Objective-C (or Swift) don't think of this as an override, so this
document shouldn't use that language. (Thanks, Varun!)

jrose-apple commented:
@swift-ci Please smoke test

jrose-apple commented:
@swift-ci Please smoke test macOS

jrose-apple commented:
@swift-ci Please smoke test macOS

jrose-apple commented:
@swift-ci Please smoke test macOS
